packages/contracts-bedrock/src/deploy-utils.ts

```diff
@@ -373,6 +394,61 @@ export const doStep = async (opts: {
   await opts.checks()
 }
 
+/**
+ * Mini helper for executing a given phase.
+ *
+ * @param opts Options for executing the step.
+ * @param opts.isLiveDeployer True if the deployer is live.
+ * @param opts.SystemDictator SystemDictator contract.
+ * @param opts.step Step to execute.
+ * @param opts.message Message to print before executing the step.
+ * @param opts.checks Checks to perform after executing the step.
+ */
+export const doPhase = async (opts: {
+  isLiveDeployer?: boolean
+  SystemDictator: ethers.Contract
+  phase: number
+  message: string
+  checks: () => Promise<void>
+}): Promise<void> => {
+  if (!(await isStartOfPhase(opts.SystemDictator, opts.phase))) {
```

Comment:
nit: this code is much more difficult to read than it needs to be, nesting `await` in expressions is an antipattern

```js
const isStart = await isStartOfPhase(...)
if (!isStart) { ... }
```

Reply:
Good call out. It followed the same pattern used above in `isStep()`, but I'll fix both for consistency.